classes/migration/CommonMigration.inc.php

```diff
@@ -88,10 +88,21 @@ public function up() {
 			$table->smallInteger('disabled')->default(0);
 			$table->text('disabled_reason')->nullable();
 			$table->smallInteger('inline_help')->nullable();
-			$table->unique(['username'], 'users_username');
-			$table->unique(['email'], 'users_email');
 		});
-
+		
+		switch (Capsule::connection()->getDriverName()) {
+			case 'mysql':
+				Capsule::schema()->table('users', function (Blueprint $table) {
+					$table->unique(['username'], 'users_username');
+					$table->unique(['email'], 'users_email');
+				});
+				break;
+			case 'pgsql':
```

Comment:
This will cause problems if there were previously registrations that had mixed cases. The pre-flight check is helpful for this; I'd recommend flagging any cases where there will be collisions in the pre-flight check, and asking the user to resolve them (e.g. with "merge users") *before* running the upgrade.

Reply:
[sorry, that comment is more useful for `main`, not `stable-3_3_0`!)